### PR TITLE
Improve formatting and stop hardcoding thresholds

### PR DESCRIPTION
**Pull request by tunetheweb**
Pulling these out of #177 so that's easier to review

Our presentation of our numbers is wildly inconsistent:

image

- 2 decimal places for LCP in HUD and popup
- seconds in HUD and popup but milliseconds in console logging
- space before the units in HUD and console but not in popup
- 3 decimal places for CLS in HUD and popup even though only ever given to 2 decimal places
- 2 decimal places for INP/FID in HUD and popup as if ms precision wasn't enough
- Popup uses user's locale, others do not
- Popup as a metric on threshold (e.g. INP 200ms) fails when others say it's got to be > 200ms (this is why it's bad to have thresholds repeated all over the place!)
- HUD doesn't use amber but rest of them do

This PR cleans that all up:

image

### CODE REVIEW

**Review by rviscomi — APPROVED** [review]
Very nice!

File: src/browser_action/vitals.js [hunk 1]
```diff
@@ -57,6 +53,16 @@
...
   });
 
+  function toLocaleFixed({value, unit, precision }) {
+    return value.toLocaleString(undefined, {
+      style: unit && 'unit',
+      unit,
+      unitDisplay: 'short',
+      minimumFractionDigits: precision ?? DEFAULT_UNITS_OF_PRECISION,
+      maximumFractionDigits: precision ?? DEFAULT_UNITS_OF_PRECISION
+    });
```
Comment (brendankenny) [inline src/browser_action/vitals.js:63]: Might want to double check this doesn't suffer from the same issue as #107. String formatters can be annoyingly slow to initialize and this runs in the main thread.

Comment (tunetheweb) [reply src/browser_action/vitals.js:63]: Urgh. This is why we can't have nice things!! OK might revert this change (but in Console, HUD and overlay).

Comment (brendankenny) [reply src/browser_action/vitals.js:63]: Other ideas: Log during idle time? Create the formatters once and save them (they may be cached by the browser and this is already functionally happening as in #107).

Comment (tunetheweb) [reply src/browser_action/vitals.js:63]: Given only the pop up showed this before and the console log and HUD didn't show it using locale, I'm tempted to just manually format this `X.XXX s`, `XXX ms` or `X.XX`. It's sad that we can't easily support people's preferred locales but I still think consistency and performance is better.

Comment (tunetheweb) [reply src/browser_action/vitals.js:63]: https://github.com/GoogleChrome/web-vitals-extension/pull/179

**Pull request merged**